dataeng/jobs/AnalyticsEmailOptin.groovy

```diff
@@ -0,0 +1,182 @@
+            }
+        }
+    }
+
+    wrappers {
+        timestamps()
+        buildName('#${BUILD_NUMBER} ${ENV,var="ORG"}')
+    }
+
+    steps {
+        shell(readFileFromWorkspace("dataeng/resources/email-optin-worker.sh"))
+    }
+
+    publishers {
+        textFinder("Task OrgEmailOptInTask failed fatally", '', true, false, false)
+    }
+}
+
+job ('analytics-email-optin-master-dsl-test') {
+
+    parameters{
+        stringParam('ORGS','*', 'Space separated list of organizations to process. Can use wildcards. e.g.: idbx HarvardX')
+        stringParam('EXPORTER_BRANCH','environment/production',
+                'Branch from the edx-analytics-exporter repository. For tags use tags/[tag-name]. Should be environment/production.')
+        stringParam('PLATFORM_BRANCH','zafft/analytics-exporter-settings-hotfix',
+                'Branch from the edx-platform repository. For tags use tags/[tag-name]. Should be release.')
+        stringParam('SECURE_BRANCH','release',
+                'Branch from the analytics-secure repository, where the configuration settings reside. For tags use tags/[tag-name]')
+        stringParam('CONFIG_FILENAME','default.yaml', 'Name of configuration file in analytics-secure/analytics-exporter.')
+        stringParam('OUTPUT_BUCKET', commonConfigMap.get('EMAIL_OPTIN_OUTPUT_BUCKET'), 'Name of the bucket for the destination of the email opt-in data.')
+        stringParam('OUTPUT_PREFIX','email-opt-in-', 'Optional prefix to prepend to output filename.')
+        stringParam('NOTIFICATION_EMAILS', commonConfigMap.get('EXTENDED_NOTIFY_LIST'),
+                'Space separated list of emails to notify in case of failure.')
+        stringParam('DATE_MODIFIER','',
+                'Used to set the date of the CWSM dump.  Leave blank to use today\'s date.  Set to "-d 201x-0x-0x" if that is when the CWSM dump took place, typically the preceding Sunday.  (Leave off quotes.)')
+        stringParam('ORG_CONFIG','data-czar-keys/config.yaml', 'Path to the data-czar organization config file')
+        stringParam('DATA_CZAR_KEYS_BRANCH','master', 'Branch of the Data-czar-keys repository to use')
+    }
+
+    multiscm{
+        git {
+            remote {
+                url('git@github.com:edx/edx-platform.git')
+                branch('$PLATFORM_BRANCH')
+                credentials('1')
+            }
+            extensions {
+                pruneBranches()
+                relativeTargetDirectory('edx-platform')
+            }
+        }
+        git {
+            remote {
+                url('git@github.com:edx/edx-analytics-exporter.git')
+                branch('$EXPORTER_BRANCH')
+                credentials('1')
+            }
+            extensions {
+                pruneBranches()
+                relativeTargetDirectory('analytics-exporter')
+            }
+        }
+        git {
+            remote {
+                url(commonConfigMap.get('SECURE_REPO_URL'))
```

Comment:
So this was my question from earlier this morning -- I was asking whether these would just be job parameters or something else.   $SECURE_BRANCH is a job parameter (on the next line), so I thought this could be as well.  Then it is up to the place that defines the actual parameters to pass to the job to provide these URLs as well.    So now I think I see that there are no job parameters being passed in when these define jobs that are automatically scheduled, so the values used in the parameter defaults have to come from some external location at job definition time instead of job runtime.  The purpose of the parameters, then, is to provide customization for when running jobs when they are *not* scheduled.  So, we get one file to define common variable values, pointed to by the environment variable COMMON_VARS.    

I guess in future, we could load a second file that was more customized if we found that we needed it.   The same global logic, but we would read COMMON_VARS_DIR, append an "analytics-email-optin-vars.yaml" path to the end, and try to read that if it exists, into say jobConfigMap.   Then have a method that looks in both commonConfigMap and jobConfigMap to get values, looking at jobConfigMap for overrides.   We might find that it's easier than shoving everything into one file with lots of per-job values.

Reply:
I could make this a parameter and provide the default. I just made it exactly like what we have on the admin box right now. I was thinking along the same lines that we would have one common file and if we needed more customization we would have yaml files per job.

Reply:
Sorry, that wasn't my intention.  This doesn't need to be a parameter since we don't expect anyone running this job manually to ever need to change that value.  My comment was more about customization in future with yaml files per job.   It sounds like we agree on that.

Reply:
Alright, so @azafty468 and I had a discussion and we agreed on having a COMMON_VARS_DIR parameter in the seed job. The individual DSL scripts would then append the yaml file name to the path(as per your suggestion) and read in the secret stuff for configuration the individual jobs. We can then add more yaml files on a per job basis(and have a common.yaml one like we do now) without needing to change the seed job. Does that seem appropriate ?